hbase-common/src/main/java/org/apache/hadoop/hbase/util/UnsafeAvailChecker.java

```diff
@@ -21,50 +21,64 @@
 import java.lang.reflect.Method;
 import java.security.AccessController;
 import java.security.PrivilegedAction;
+import sun.misc.Unsafe;
 
 import org.apache.yetus.audience.InterfaceAudience;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@SuppressWarnings("restriction")
 @InterfaceAudience.Private
 public class UnsafeAvailChecker {
 
+  private static final Unsafe UNSAFE;
```

Comment:
Why this change? I think one of the purpose of this class is to not reference Unsafe directly?

Reply:
Indeed that was the old reasoning. The idea was that not all JVM implementations have an `Unsafe` class, so interfacing with it, if it existed, had to be done entirely via reflection.